src/schema.graphql

```diff
@@ -201,6 +202,12 @@ type Query {
   getLevels: [Int]
   getLimits: Limits
   getWalletFees: WalletFees
+  generate2fa: TwoFactorSecret
```

Comment:
Shouldn't this be under `Mutation`? `Generate` implies a side effect. `Queries` should be side effect free.

Reply:
There's no side effect. Just returns a new `secret` but does not interact with data in any way.

Reply:
What is the source of the new secret? Is anything persisted? Anything with the word `generate` implies a creation (which in GQL is handled via the `Mutation` path). If we are not creating something perhaps the word is wrong.

Reply:
I would argue that when it comes to generating crypto-related keys/secrets, there's typically no side effect with functions that have `generate` in their name. For example: the `generateKeyPair` function from node's inbuilt crypto module comes to mind. 
Having said that I do agree that something like `generateNew2FASecret` would be a better name

Reply:
we don't persist anything yet at this time of the call. we create/pass a secret that the client will send back to the server once the client verified within the mobile that he has saved it by providing a 6 digits associated to the secret.

I don't think not saving the temporary secret can be a source of attack but maybe something to consider

Reply:
> for example: the generateKeyPair function from node's inbuilt crypto module comes to mind.

This may be generally true in the context of crypto-functions but GraphQL has its own conventions and here we are designing a public GQL interface. Not a crypto-library API.

I'm not sure what the idiomatic way to handle this in GQL would be and the GQL layer needs reworking anyway so as long as nothing is persisted we can probably leave it like this for the time being.

Another thing to consider is that `Queries` can typically be combined with various caching layers which would break here because each call returns a different response.

so TLDR - I believe its probably a `Mutation` but not a big deal to leave it as is for now. Needs revisiting when we re-design the whole GQL layer.

Reply:
> Another thing to consider is that Queries can typically be combined with various caching layers which would break here because each call returns a different response.

this is a very valid point. we're not caching anything at the GQL layer yet but we should at some point

Reply:
the counter argument is that even if we were to return a cached secret / user, this would not be issue I guess (as long as the secret being generated is obviously different for each user)

Reply:
IMO, a non-idempotent operation should be a mutation.